src/makielayout/blocks/polaraxis.jl

```diff
@@ -0,0 +1,537 @@
+        color = po.θgridcolor,
+        linestyle = po.θgridstyle,
+        linewidth = po.θgridwidth,
+        visible = po.θgridvisible
+    )
+    # minor grids
+    rminorgridplot = lines!(
+        po.blockscene, rminorgridpoints;
+        color = po.rminorgridcolor,
+        linestyle = po.rminorgridstyle,
+        linewidth = po.rminorgridwidth,
+        visible = po.rminorgridvisible
+    )
+
+    θminorgridplot = lines!(
+        po.blockscene, θminorgridpoints;
+        color = po.θminorgridcolor,
+        linestyle = po.θminorgridstyle,
+        linewidth = po.θminorgridwidth,
+        visible = po.θminorgridvisible
+    )
+    # tick labels
+    rticklabelplot = text!(
+        po.blockscene, rtick_pos_lbl;
+        textsize = po.rticklabelsize,
+        font = po.rticklabelfont,
+        color = po.rticklabelcolor,
+        align = (:left, :bottom),
+        space = :pixel,
+        markerspace = :pixel
+    )
+
+    θticklabelplot = text!(
+        po.blockscene, θtick_pos_lbl;
+        textsize = po.θticklabelsize,
+        font = po.θticklabelfont,
+        color = po.θticklabelcolor,
+        align = (:center, :center),
+        space = :pixel,
+        markerspace = :pixel
+    )
+
+    clippoints = lift(spinepoints) do spinepoints
+        area = pixelarea(po.scene)[]
+        ext_points = Point2f[
+            (left(area), bottom(area)),
+            (right(area), bottom(area)),
+            (right(area), top(area)),
+            (left(area), top(area)),
+        ]
+        return GeometryBasics.Polygon(ext_points, [spinepoints])
+    end
+
+    clipcolor = lift(parent(po.blockscene).theme.backgroundcolor) do bgc
+        bgc = to_color(bgc)
+        if alpha(bgc) == 0f0
+            return to_color(:white)
+        else
+            return RGBf(red(bgc), blue(bgc), green(bgc))
+        end
+    end
+
+    clipplot = poly!(
+        po.blockscene,
+        clippoints,
+        color = clipcolor,
+        space = :pixel,
+        strokewidth = 0,
+        visible = po.clip,
+    )
+
+    translate!.((spineplot, rgridplot, θgridplot, rminorgridplot, θminorgridplot, rticklabelplot, θticklabelplot), 0, 0, 100)
+    translate!(clipplot, 0, 0, 99)
+
+    return (spineplot, rgridplot, θgridplot, rminorgridplot, θminorgridplot, rticklabelplot, θticklabelplot)
+
+end
+
+function calculate_polar_title_position(area, titlegap, align, θaxisprotrusion)
+    local x::Float32 = if align === :center
+        area.origin[1] + area.widths[1] / 2
+    elseif align === :left
+        area.origin[1]
+    elseif align === :right
+        area.origin[1] + area.widths[1]
+    else
+        error("Title align $align not supported.")
+    end
+
+    # local subtitlespace::Float32 = if ax.subtitlevisible[] && !iswhitespace(ax.subtitle[])
+    #     boundingbox(subtitlet).widths[2] + subtitlegap
+    # else
+    #     0f0
+    # end
+
+    local yoffset::Float32 = top(area) + titlegap + θaxisprotrusion.top #=+
+        subtitlespace=#
```

Comment:
Are the `local` doing anything here?

Reply:
No, copied wholesale from the implementation for Axis.  I'll go through and clean that up tomorrow.